src/node.js

```diff
@@ -717,6 +717,12 @@
         stdin._handle.readStop();
       });
 
+      stdin.isTTY = true;
+      stdin.isRaw = false;
+      stdin.setRawMode = function setRawMode() {
+        throw new Error('Not a raw device');
+      };
```

Comment:
Yeah, `stdin.isTTY` should definitely be false in the PIPE case. I corrected that in my latest diff suggestion.

The whole change is more of a correctness thing. Most scripts do just `if (stdin.isTTY)`, in which case it wouldn't matter if it's undefined or false.